### DISCUSSION TITLE
Crypto requirements - key rotation

### DISCUSSION THREAD

**Discussion by lwierzbicki** [OWASP/masvs#579] [Q&A] [answered]
During reading crypto requirements, I cannot find requirement which is related to key rotation (changing key and its related and appriopate time length when key is used). I started to consider this from key lifecycle and I think it is missing here.

Related guide [NIST SP800-175B chapter 5.1.1](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-175Br1.pdf)

**Comment by commjoen** [comment]
Good point: either we extend MSTG-CRYPTO-5 or create a new requirement 👍

**Comment by vixentael** [comment]
+1 to key rotation. 

@julepka and I recommend the following requirement (see https://github.com/OWASP/owasp-masvs/pull/500/):

```
| 3.7 | MSTG-CRYPTO-7 | Each cryptographic key is regularly rotated depending on its properties and cryptosystem design. |   | ✓ |
```

**Comment by cpholguera** [comment, marked as answer]
This is considered in the new release of MASVS-CRYPTO (#612, see [MASVS-CRYPTO-3](https://docs.google.com/spreadsheets/d/1oyVMntRLaLHMWNfQqrWk3e0Q_HUI0Yo1opRB_hyYrpY/edit#gid=0&range=L25)). Thanks a lot @lwierzbicki!

**Comment by cpholguera** [comment]
Now covered in this to-be-created weakness 

https://mas.owasp.org/MASWE/MASVS-CRYPTO/MASWE-0011/

@lwierzbicki, @vixentael would you like to create this item for the new MASWE?
